paymentsheet/src/test/java/com/stripe/android/paymentsheet/DeferredIntentValidatorTest.kt

```diff
@@ -0,0 +1,223 @@
+package com.stripe.android.paymentsheet
+
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.model.SetupIntentFixtures
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.paymentsheet.PaymentSheet.IntentConfiguration
+import com.stripe.android.testing.PaymentIntentFactory
+import org.junit.Assert.assertThrows
+import kotlin.test.Test
+
+@OptIn(ExperimentalPaymentSheetDecouplingApi::class)
+internal class DeferredIntentValidatorTest {
+
+    @Test
+    fun `Fails if PaymentIntent is validated against IntentConfiguration in setup mode`() {
+        val paymentIntent = PaymentIntentFactory.create()
+        val intentConfiguration = makeIntentConfigurationForSetup()
+
+        val failure = assertThrows(IllegalArgumentException::class.java) {
```

Comment:
kotlin test has a variant that allows a generic version of this. https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-fails-with.html